test/libsolidity/syntaxTests/abiEncoder/passing_memory_to_interface_that_expects_calldata_should_not_throw.sol

```diff
@@ -0,0 +1,10 @@
+contract Sender {
```

Comment:
I think this should be more thoroughly tested. The reporter provided one case but when implementing a fix, please always try to come up with other similar cases and add some of them. For example here:
- I'd test all the 6 combinations (`calldata`/`calldata`, `memory`/`calldata`, `calldata`/`memory`, `memory`/`memory`, `storage`/`calldata`, `storage`/`memory`). The last one is probably covered by existing tests but please check.
- Are there any other types that could cause a mismatch like this?
    - So far I managed to come up with three (see https://github.com/ethereum/solidity/issues/12718#issuecomment-1064322093). Please add these as test cases here and check if your fix already covers them. Also, please try to see if you can come up with more.
    - For a good measure I'd still add something some more types in this case: `bytes`, static and dynamic array, struct.
- A semantic test for the `calldata`/`memory` case would not hurt - so that we're sure that codegen can actually handle it correctly despite the slight difference in types. A function pointer case might also warrant a semantic test since I remember that we did have some asserts failing at some point because they did not expect slightly different pointer types. I'd verify it works at least.

Reply:
Could you add some of these?